### PR TITLE
Count ElseIf as executable line metric for VB.NET

### LINKED ISSUE #5109: Count `ElseIf` as executable line metric for VB.NET
VB.NET `ElseIf` line should be considered as executable in metrics.
```
ElseIf Count = 1 Then
```

It should behave the same as C# variant
```
else if (Count == 1)
{
```

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #5109

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SnippetCompilerTest.cs [hunk 1]
```diff
@@ -121,8 +121,8 @@ public class Sample
 // Error CS1513 } expected
 {";
             using var log = new LogTester();
-            Action a = () => new SnippetCompiler(code);
-            a.Should().Throw<InvalidOperationException>();
+            Func<SnippetCompiler> f = () => new SnippetCompiler(code);
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SnippetCompilerTest.cs:124]: Codesmell fix from previous PR https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&pullRequest=5082&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SnippetCompilerTest.cs [hunk 2]
```diff
@@ -121,8 +121,8 @@ public class Sample
...
+            f.Should().Throw<InvalidOperationException>();
```
Comment (Corniel) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SnippetCompilerTest.cs:125]: ctor (or `@constructor`) is a better name.

**Review by Corniel — APPROVED** [review]
LGTM (just one rename suggestion).

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**